File: latihan2/python/Product.py
#Saya Mohammad Ray Mosaid mengerjakan Latihan 2 dalam mata kuliah DPBO
#untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan.
#Aamiin.

#Memeanggil class Product
class Product:

    #variable/atribut private
    __price = 0
    __idProduct = ""

    #membuat konsruktor untuk mengubah/mengakses variabel/atribut private
    def __init__(self, price, idProduct):
        self.__price = price
        self.__idProduct = idProduct
    
    #membuat fungsi untuk mengeset variabel/atribut private
    def setprice(self, price):
        self.__price = price
    def setidProduct(self, idProduct):
        self.__idProduct = idProduct
        
    #membuat fungsi untuk me-return hasil
    def getprice(self):
        return self.__price
    def getidProduct(self):
        return self.__idProduct
    

File: latihan2/c++/product.cpp
/*
    Saya Mohammad Ray Mosaid mengerjakan Latihan 2 dalam mata kuliah DPBO
    untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan.
    Aamiin.
*/

//memanggil header yang dibutuhkan
#include <iostream>
#include <string.h>

using namespace std;

//membuat class Product sebagai parent dari seluruh proses inheritance
class product{
    //membuat atribut private
	private:
        int price;
        string idProduct;

    public:
        //membuat konstruktor
        product(){
        }
        //membuat fungsi untuk mengeset atribut private
        void setprice(int price){
            this->price = price;
        }
        void setidProduct(string idProduct){
            this->idProduct = idProduct;
        }

        //membuat fungsi untuk me-return atribut private
        int getprice(){
            return price;
        }
        string getidProduct(){
            return idProduct;
        }

        //membuat destruktor
        ~product(){
        }
};
